klog.go

```diff
@@ -874,8 +881,20 @@ func (sb *syncBuffer) Sync() error {
 	return sb.file.Sync()
 }
 
+//IsFileReachedMaxSize checks whether the file exceeds the max size limitation.
+func IsFileReachedMaxSize(bufferSize uint64) bool {
+	if logging.logFile != "" {
+		// Flag logFileMaxSize is in MB for user convenience.
+		maxSize := logging.logFileMaxSize * 1024 * 1024
+		// If logFileMaxSize is zero, we don't have limitations on the log size.
+		return maxSize > 0 && bufferSize >= maxSize
+	}
+	// If "log_file" flag is not specified, the log files will be cleaned up when reaches a fixed size.
+	return bufferSize >= MaxSize
+}
```

Comment:
I would just precompute a max size to simplify this check, as it needs to be performed on every log write. You can substitute the "unlimited" value with `math.MaxUint64`